Logger.md

```diff
@@ -0,0 +1,107 @@
+# Logger
+
+`Logger` is an interface that helps decouple client classes from concrete implementations of a logging system.
+
+## Declaration
+
+```swift
+enum LogLevel {
+    case info
+    case warning
+    case error
+}
+
+protocol Logger {
+  func log(_ level: LogLevel, tag: String, message: String)
+  func print(tag: String, message: String)
+  func warning(tag: String, message: String) 
+  func error(tag: String, message: String)
```

Comment:
I would not define the methods `print`,  `warning`, and `error in the main definition of the protocol. I would list them as extensions directly as these methods are not supposed to be re-implemented.

Reply:
The current protocol extension is a convenience for the most common scenario. However, I wouldn't like to force concrete implementations in any way. If they wanted to override, they should be able to.

Reply:
by providing these 3 methods we are already providing and enforcing a specific implementation of the methods. I don't foresee any situation in which you want to override one of these three methods but not the main log one. 

As you define only 3 log levels (and this can't be modified), you can also provide 3 main logging methods that forward to the main one, the one that must be implemented by the Logger. That's why I was thinking it is better to leave the main `log` function alone as the main requirement to be implemented and let the other 3 methods as additional things.  Note that the final user can still override these methods anyway as Swift allows it, but at least we are not directly asking him to "please implement them" by grouping it in different sections of the file.

Any thoughts the others?

Reply:
Ok, I don't feel that strongly about this. I've changed it. 

Just a note. Methods in a protocol extension can only be overwritten if the receiver is not declared as Protocol type. If they are, the implementation of the protocol extension is used regardless of the what the conforming class does. So in this case, the design of the protocol will force conforming classes depending on how they are used.

Reply:
Very good point. We can do whatever you think is better